<html>
<head>
	<title>USeq Applications</title>

<style type="text/css">
	#cr{text-align:center; color: #000000; font-weight: bold}
	#rt{text-align:right; color: #000000; font-weight: bold}
	#grBk {background-color: #CC9966;}
	TD {font-family: Verdana, Arial, Helvetica, sans-serif; font-size:12;}
	H1 {color: #996633; font:arial; font-size:16;}
	H2 {color: #996633; font:arial; font-size:12;}
	BODY {color:black; background-color:white; font-family: Verdana, Arial, Helvetica, sans-serif; font-size:12;}
	
	A:link    {text-decoration: none; color: #000000; font-weight: bold}  
	A:visited {text-decoration: none; color: #000000; font-weight: bold}   
	A:hover   {text-decoration: none; color: #FFCC66; font-weight: bold} 
	A:active  {text-decoration: none; color: #000000; font-weight: bold}   
</style>
</head>

<body>
<center>
<H1>Core Applications</H1>

Launch these applications using the -jar option, for example, 'java -jar pathTo/USeq/Apps/FileSplitter' .<br>Click the name to see the command line menu.<p>
		
<table ID='grBk' width="750" cellspacing="5" cellpadding="2" >	

<tr><td ID='cr' COLSPAN="2">ChIP-Seq Apps</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ChIPSeq">ChIP Seq</a><td>Wraps a variety of the following USeq apps to run a differential treatment vs control/ input ChIP-Seq analysis. It parses raw alignment files, filters duplicate reads, generates <a href="cmdLnMenus.html#ReadCoverage">ReadCoverage</a> tracks, estimates the peak shift using the <a href="cmdLnMenus.html#PeakShiftFinder">PeakShiftFinder</a>, and lastly, runs the <a href="cmdLnMenus.html#ScanSeqs">MultipleReplicaScanSeqs</a> and the <a href="cmdLnMenus.html#EnrichedRegionMaker">EnrichedRegionMaker</a> to identify ChIP-Seq peaks.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#QCSeqs">QC Seqs</a><td>QCSeqs takes directories of chromosome specific PointData xxx.bar.zip files that represent replicas of signature sequencing data, merges the strands, uses a sliding window to sum the hits, and calculate Pearson correlation coefficients for the window sums between each pair of replicas.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#PeakShiftFinder">Peak Shift Finder</a><td>Scans stranded point data for a skew in the distribution of peaks, a signature of a real chIP-seq peak.  Calculates the bp difference.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ScanSeqs">Scan Seqs</a><td>Takes stranded chromosome specific PointData and uses a sliding window to calculate several smoothed window statistics. These include a binomial p-value, a q-value FDR, an empirical FDR, and a Bonferroni corrected binomial p-value for flagging peak shift strand skew. </tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#MultipleReplicaScanSeqs">Multiple Replica Scan Seqs</a><td>MRSS uses a sliding window and Ander's <a href='http://www-huber.embl.de/users/anders/DESeq/'>DESeq</a> negative binomial pvalue -> Benjamini & Hochberg FDR statistics to identify enriched and reduced regions in a genome. Both treatment and control PointData sets are required, one or more biological replicas. MRSS generates window level differential count tracks for the FDR and normalized log2Ratio as well as a binary window object xxx.swi file for downstream use by the EnrichedRegionMaker. Lastly, MRSS also makes use of DESeq's variance corrected count data to cluster your biological replicas. </tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#EnrichedRegionMaker">Enriched Region Maker</a><td>Combines scored windows from ScanSeqs into larger Enriched Regions/ Binding Peaks given one or more thresholds. Can also be used to find the best peak within each Enriched Region.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ScoreEnrichedRegions">Score Enriched Regions</a><td>Determines if a set of regions specified by the user is more or less enriched than a randomly generated set of regions matched on chromosome, region and GC Content.  The program checks each region individually and the dataset as a whole.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#Simulator">Simulator</a><td>Generates simulated chIP-seq reads for alignment to a reference.</tr>

<tr><td ID='cr' COLSPAN="2">&nbsp;</tr>
<tr><td ID='cr' COLSPAN="2">RNA-Seq Apps</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#RNASeq">RNA Seq</a><td>Wraps a variety of the following USeq apps to run a differential treatment vs control gene expression RNA-Seq analysis. It parses raw alignment files, generates <a href="cmdLnMenus.html#ReadCoverage">ReadCoverage</a> tracks, runs the <a href="cmdLnMenus.html#MultipleReplicaScanSeqs">MultipleReplicaScanSeqs</a> and the <a href="cmdLnMenus.html#EnrichedRegionMaker">EnrichedRegionMaker</a> to identify novel transfrags, and identifies differential gene expression and splicing using the <a href="cmdLnMenus.html#MultipleReplicaDefinedRegionScanSeqs">MultipleReplicaDefinedRegionScanSeqs</a>.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#DefinedRegionDifferentialSeq">Defined Region Differential Seq</a><td>DRDS takes bam files representing a treatment and control experiment or a multiple condition/ time series and identifies differentially expressed genes under any pairwise comparison using Simon Anders DESeq. Alternative splicing is estimated using a chi-square test of independence.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#DRDSAnnotator">DRDS Annotator</a><td>This application annotates DefinedRegionDifferentialSeq xlsx files using Ensembl biomart tab-delimited annotation files.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#Telescriptor">Telescriptor</a><td>Compares two RNASeq datasets for possible telescripting. Generates a spreadsheet of statistics for each gene as well as a variety of graphs in exonic bp space</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#AllelicExpressionDetector">Allelic Expression Detector</a><td>Application for identifying allelic expression based on a table of snps and bam alignments that have been filtered for alignment bias.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#MiRNACorrelator">MiRNA Correlator</a><td>Generates a spreadsheet to use in comparing changing miRNA levels to changes in gene expression.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#MakeTranscriptome">Make Transcriptome</a><td>Makes all known and theoretical splice junctions and transcripts from a table of gene models. Read through occurs with small exons to the next up or downstream. Include the splices fasta file along with the chromosome files when building a genome index for RNA-Seq alignments.</tr>	
<tr><td ID='rt'><a href="cmdLnMenus.html#RNASeqSimulator">RNASeq Simulator</a><td>RSS takes novoalignments from RNA-Seq data and simulates over dispersed, multiple replica, differential, non-stranded RNA-Seq datasets.</tr>

<tr><td ID='cr' COLSPAN="2">&nbsp;</tr>
<tr><td ID='cr' COLSPAN="2">RNA Editing Apps</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#InosinePredict">Inosine Predict</a><td>IP estimates the likelihood of ADAR RNA editing using the multiplicative 4L,4R model described in Eggington et. al. 2010.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#RNAEditingPileUpParser">RNA Editing PileUp Parser</a><td>Parses a SAMTools mpileup output file for refseq A bases that show evidence of RNA editing via conversion to Gs, stranded. Base fraction editing is calculated for bases passing the thresholds for viewing in IGB and subsequent clustering with the RNAEditingScanSeqs app</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#RNAEditingScanSeqs">RNA Editing ScanSeqs</a><td>RESS attempts to identify clustered editing sites across a genome using a sliding window approach.  Each window is scored for the pseudomedian of the base fraction edits as well as the probability that the observations occured by chance using a permutation test based on the chiSquare goodness of fit statistic.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#DefinedRegionRNAEditing">Defined Region RNA Editing</a><td>DRRE scores regions for the pseudomedian of the base fraction edits as well as the probability that the observations occured by chance using a permutation test based on the chiSquare goodness of fit statistic</tr>

<tr><td ID='cr' COLSPAN="2">&nbsp;</tr>
<tr><td ID='cr' COLSPAN="2">Sam Alignment Apps</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#SamTranscriptomeParser">Sam Transcriptome Parser</a><td>Takes SAM alignment files that were aligned against chromosomes and extended splice junctions (see MakeTranscriptome app) and converts the coordinates to genomic space.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#Sam2USeq">Sam 2 USeq</a><td>Generates per base read depth stair-step xxx.useq graph files for visualization in IGB from SAM alignment files. Options are available for generating relative and stranded graphs.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#SamSVFilter">Sam SV Filter</a><td>Filters SAM records based on their intersection with a list of target regions for structural variation analysis. Paired alignments are kept if they align to at least one target region. These are split into those that align to different targets (span), the same target with sufficient softmasking (soft), or one target and somewhere else(single).</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#MergePairedAlignments">Merge Paired Alignments</a><td>Merges proper paired alignments that pass a variety of checks and thresholds. Usefull for avoiding non-independent variant observations and other double counting issues when reads overlap.</tr>	
<tr><td ID='rt'><a href="cmdLnMenus.html#AlignmentEndTrimmer">Alignment End Trimmer</a><td>This application can be used to trim alignments according to the density of mismatches.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#SamAlignmentExtractor">Sam Alignment Extractor</a><td>Parses all of the intersecting sam alignments to a given bed file of genomic regions.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#SamComparator">Sam Comparator</a><td>Compares coordinate sorted, unique, alignment sam/bam files.  Splits alignments with the same name into those that match chrom and position or mismatch. Use to remove alignment bias for allelic expression.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#SamReadDepthSubSampler">Sam Read Depth Sub Sampler</a><td>Filters, randomizes, subsamples each coordinate sorted bam alignment file to a target base level read depth. Useful for reducing extreem read depths over localized areas.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#CompareParsedAlignments">Compare Parsed Alignments</a><td>Compares two parsed alignments for a common distribution of snps using R's Fisher's Exact. Run the ParseIntersectingAlignments with the same snp table first.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#SamSubsampler">Sam Subsampler</a><td>Filters, randomizes, subsamples and sorts sam/bam alignment files.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#CalculatePerCycleErrorRate">Calculate Per Cycle Error Rate</a><td>Calculates per cycle error rates provided a sorted indexed bam file and a fasta sequence file.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#SamParser">Sam Parser</a><td>Parses SAM and BAM files into center position binary PointData xxx.bar files.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#MergeSams">MergeSams</a><td>Merges SAM/BAM files, uses a consensus stripped header if one isn't provided.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#SamSplitter">SamSplitter</a><td>Splits a queryName sorted BAM in half, randomly assigns read groups to either.</tr>


<tr><td ID='cr' COLSPAN="2">&nbsp;</tr>
<tr><td ID='cr' COLSPAN="2">VCF Apps</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#MultiSampleVCFFilter">Multi Sample VCF Filter</a><td>Splits vcf file(s) containing multiple sample records into those that pass and fail the user defined tests and sample level thresholds.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#VCFComparator">VCF Comparator</a><td>Compares test vcf file(s) against a gold standard key of trusted vcf calls.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#VCFAnnotator">VCF Annotator</a><td>Adds a variety of ANNOVAR annotations to the INFO field of each vcf record.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#VCFSpliceAnnotator">VCF Splice Annotator</a><td>Scores vcf variants for gain or loss of splice junctions using the MaxEntScan algorithms.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#VCFReporter">VCF Reporter</a><td>Exports particular VCF record INFO to a spreadsheet or secondary VCF file.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#VCFTabix">VCFTabix</a><td>Converts vcf files to a SAMTools compressed vcf tabix format.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#VarScanVCFParser">VarScanVCFParser</a><td>Removes non SOMATIC calls and replace QUAL score with the ssc score.</tr>

<tr><td ID='cr' COLSPAN="2">&nbsp;</tr>
<tr><td ID='cr' COLSPAN="2">Bisulfite Sequencing & Methylation Array Apps</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#NovoalignBisulfiteParser">Novoalign Bisulfite Parser</a><td>Parses Novoalign single and paired bisulfite sequence alignment files into xxx.bed and PointData file formats. Generates several summary statistics on converted and non-converted C contexts. Flattens overlapping reads in a pair to call consensus bps.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#BisStat">Bis Stat</a><td>Takes PointData from converted and non-converted C bisulfite sequencing data parsed using the NovoalignBisulfiteParser and generates several xxCxx context statistics and graphs for visualization in IGB. Estimates whether a given C is methylated using a binomial distribution.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#BisStatRegionMaker">Bis Stat Region Maker</a><td>Takes serialized window objects from BisStat, thresholds based on the min and max fraction methylation params and prints regions in bed format meeting the criteria.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#BisSeq">Bis Seq</a><td>Takes two condition (treatment and control) PointData from converted and non-converted C bisulfite sequencing data parsed using the NovoalignBisulfiteParser and identifies differential methylated regions using either a fisher exact or a chi-square test p-values converted to B&H FDRs.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#DefinedRegionBisSeq">Defined Region Bis Seq</a><td>Takes two condition (treatment and control) PointData from converted and non-converted C bisulfite sequencing data parsed using the NovoalignBisulfiteParser and scores regions defined by the user for differential methylated using either a fisher exact or a chi-square test p-values converted to B&H FDRs.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#StrandedBisSeq">Stranded Bis Seq</a><td>Looks for strand bias in CG methylation from one dataset using fischer/ chi-square tests followed by a Benjamini and Hochberg FDR correction. <b>WARNING:</b> many bisulfite datasets display strand bias due to preferential breakage of C rich strands.  Use this app with caution.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ScoreMethylatedRegions">Score Methylated Regions</a><td>For each region fetches the underlying methylation data. A p-value for each region's fraction methylated as well as a fold enrichment over random can be calculated using randomly drawn regions matched by chromosome, region length, # obs, and GC content.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ParsePointDataContexts">Parse PointData Contexts</a><td>Parses PointData for particular 5bp genomic sequence contexts.  Useful for splitting data into CG and nonCG contexts.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#BisSeqAggregatePlotter">BisSeq Aggregate Plotter</a><td>BSAP merges bisulfite data over defined regions to generate data for class average agreggate plots of fraction methylation.</tr>			
<tr><td ID='rt'><a href="cmdLnMenus.html#BisSeqErrorAdder">Bis Seq Error adder</a><td>Takes PointData from the NovoalignBisulfiteParser and simulates a worse non-coversion rate by randomly picking converted observations and making them non-converted.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#AllelicMethylationDetector">Allelic Methylation Detector</a><td>AMD identifies regions displaying allelic methylation, e.g. ~50% average mCG methylation yet individual read pairs show a bimodal fraction distribution of either fully methylated or unmethylated.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#MethylationArrayScanner">Methylation Array Scanner</a><td>MAS takes paired and non-paired sample PointData representing beta values (0-1) from arrays and attempts to identify regions with enriched/ reduced signal using a sliding window approach.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#MethylationArrayDefinedRegionScanner">Methylation Array Defined Region Scanner</a><td>MADRS takes paired and non-paired sample PointData representing beta values (0-1) from arrays and scores user defined regions for differences in methylation.</tr>

<tr><td ID='cr' COLSPAN="2">&nbsp;</tr>
<tr><td ID='cr' COLSPAN="2">Fasta and Fastq Apps</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#SRAProcessor">SRA Processor</a><td>Fetchs SRA files from the Sequence Read Archive and converts them to gzipped fastq. Can be used with <a href='http://bioserver.hci.utah.edu/BioInfo/index.php/Software:Tomato'>tomato</a> to launch alignments etc. on each fastq file.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ReferenceMutator">Reference Mutator</a><td>Takes a directory of fasta chromosome sequence files and converts the reference allele to the alternate provided by a snp mapping table.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ConvertFasta2GCBarGraph">Convert Fasta 2 GC BarGraph</a><td>Converts fasta files into graph files containing a 1 over each C in a CpG context.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#MaskExonsInFastaFiles">Mask Exons In Fasta Files</a><td>Replaces exonic sequence with Ns.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#MaskRegionsInFastaFiles">Mask Regions In Fasta Files</a><td>Replaces the region (or non region) sequence with Ns.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ConvertFastaA2G">Convert Fasta A2G</a><td>Converts all the a/A's to g/G's in fasta file(s) maintaining case.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ConvertFastqA2G">Convert Fastq A2G</a><td>Converts all the sequence A's to G's, case insensitive.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#FetchGenomicSequences">Fetch Genomic Sequences</a><td>Given a file containing genomic coordinates, fetches and saves the sequence.</tr>	
<tr><td ID='rt'><a href="cmdLnMenus.html#ConvertFasta2GCBoolean">Convert Fasta 2 GC Boolean</a><td>Converts fasta sequences to GC boolean arrays for use by other applications.</tr>	
<tr><td ID='rt'><a href="cmdLnMenus.html#MakeSpliceJunctionFasta">Make Splice Junction Fasta</a><td>Depreciated, use MakeTranscriptome. MSJF creates a multi fasta file containing sequences representing all possible linear splice junctions.</tr>	
<tr><td ID='rt'><a href="cmdLnMenus.html#ConcatinateFastas">Concatinate Fastas</a><td>Concatinates a directory of fasta files into a single sequence seperated by a defined number of Ns. Use this app to create artificial chromosomes for poorly assembled genomes.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ShiftAnnotationPositions">Shift Annotation Positions</a><td>Uses the information in an xxx.shifter.txt file from the ConcatinateFastas app to shift the annotation to match the coordinates of the concatinated sequence.</tr>

<tr><td ID='cr' COLSPAN="2">&nbsp;</tr>
<tr><td ID='cr' COLSPAN="2">Region Analysis Apps</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#IntersectRegions">Intersect Regions</a><td>Performs an intersection analysis on lists of genomic regions, uses random regions matched for GC content, length, array interrogated regions, and chromosome to calculate an enrichment over random and p-value. Also generates a distance to nearest region distribution histogram.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#CompareIntersectingRegions">Compare Intersecting Regions</a><td>Compares test region file(s) against a master set of regions for intersection. Reports the results as columns relative to the master.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#FilterIntersectingRegions">Filter Intersecting Regions</a><td>Sorts a bed file into those regions that intersect or don't intersect any regions in a second bed file. Fast.</tr>	
<tr><td ID='rt'><a href="cmdLnMenus.html#MergeRegions">Merge Regions</a><td>Merges genomic regions. Good for collapsing duplicate regions for creating microarray seq capture designs.</tr>	
<tr><td ID='rt'><a href="cmdLnMenus.html#SubtractRegions">Subtract Regions</a><td>Removes regions from a genomic regions file. Good for RepeatMasking genomic regions prior to tiling.</tr>	
<tr><td ID='rt'><a href="cmdLnMenus.html#ParseIntersectingAlignments">Parse Intersecting Alignments</a><td>Parses bed alignment files for intersecting reads provided another bed file of regions of interest (e.g. alleles)</tr>	
<tr><td ID='rt'><a href="cmdLnMenus.html#FindSharedRegions">Find Shared Regions</a><td>Writes out a bed file of shared regions.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#BedStats">Bed Stats</a><td>Calculates several statistics on bed files where the name column contains a short read sequence. This includes a read length distribution and frequencies of the 1st and last bp.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#IntersectKeyWithRegions">Intersect Key With Regions</a><td>IR intersects lists of genomic regions with a key of known regions.  Generates TPR, FPR, FDR, etc at each threshold.</tr>

<tr><td ID='cr' COLSPAN="2">&nbsp;</tr>
<tr><td ID='cr' COLSPAN="2">General Analysis Apps</tr>	
<tr><td ID='rt'><a href="cmdLnMenus.html#AggregatePlotter">Aggregate Plotter</a><td>Fetches point data contained within each region, zeros the coordinates, scales, sums, and window averages the values.  Useful for generating class averages from a list of annotated regions. Use a spreadsheet app to graph the results.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#IntersectLists">Intersect Lists</a><td>Intersects two lists (of gene names) and using randomization, calculates the significance of the intersection and the fold enrichment over random.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#RankedSetAnalysis">Ranked Set Analysis</a><td>Performs an intersection analysis on lists of ranked regions creating a visual box-line-box representation as well as a rank based % intersection graph.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#CorrelationMaps">Correlation Maps</a><td>CM creates correlation maps from gene expression data to look for physical gene clusters (aka gene expression neighborhoods, chromosome territories).</tr>	
<tr><td ID='rt'><a href="cmdLnMenus.html#CorrelatePointData">Correlate PointData</a><td>Calculates a Pearson Correlation Coefficient on the values of PointData found with the same positions in the two datasets.</tr>	
<tr><td ID='rt'><a href="cmdLnMenus.html#ScoreChromosomes">Score Chromosomes</a><td>Scores a genome for hits to a transcription factor binding matrix, LLPSPM.	</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ScoreSequences">Score Sequences</a><td>Scores a multi-FASTA file of sequences for hits to a transcription factor binding matrix, LLPSPM.	</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ScoreParsedBars">Score Parsed Bars</a><td>Given a list of regions and a directory of graph data in bar format, extracts all the values under each region, calculates their mean and compares it to a random background model to generate a p-value for the associated means.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ScoreSequences">Kegg Pathway Enrichment</a><td>Looks for overrepresentation of genes from a user's list in Kegg pathways using a random permutation test.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#MaxEntScanScore3">Max Ent Scan Score3</a><td>Implementation of Max Ent Scan's score3 algorithm for human splice site detection. See Yeo and Burge 2004.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#MaxEntScanScore5">Max Ent Scan Score5</a><td>Implementation of Max Ent Scan's score5 algorithm for human splice site detection. See Yeo and Burge 2004.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#MicrosatelliteCounter">Microsatellite Counter</a><td>MicrosatelliteCounter identifies and counts microsatellite repeats in MiSeq fastq files.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#TomatoFarmer">TomatoFarmer</a><td>TomatoFarmer controls an exome analysis from start to finish.  It creates alignment jobs for each of the samples in your directory, waits for all jobs to finish and then launches metrics and variant calling jobs.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ParseExonMetrics">Parse Exon Metrics</a><td>This script runs a bunch of summary metric programs and compiles the results.  It uses R and LaTex to generate a fancy pdf as an output.</tr>

<tr><td ID='cr' COLSPAN="2">&nbsp;</tr>
<tr><td ID='cr' COLSPAN="2">PointData Manipulation Apps</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#FilterPointData">Filter Point Data</a><td>FPD drops observations from PointData that intersect a list of regions (e.g. repeats).</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#SubSamplePointData">Sub Sample Point Data</a><td>Creates a random sub sampling of Point Data.  Useful for matching treatment and control datasets for visualization.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#PointDataManipulator">Point Data Manipulator</a><td>Manipulates point data to merge datasets, merge strands, shift base positions, replace scores with 1, and sum identical positions.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#MergePointData">Merge Point Data</a><td>Efficiently merges PointData, collapsing by position and possibly strand. Identical position scores are either summed or converted into counts</tr>

<tr><td ID='cr' COLSPAN="2">&nbsp;</tr>
<tr><td ID='cr' COLSPAN="2">UCSC Gene Table Apps</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#FindNeighboringGenes">Find Neighboring Genes</a><td>FNG takes a list of genes in UCSC Gene Table format and intersects them with a list of regions finding the closest gene to each region as well as all of the genes that fall within a given neighborhood. </tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ExportExons">Export Exons</a><td>Takes a UCSC gene table and exports the exons to a bed file +/- a bp buffer.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ExportIntergenicRegions">Export Intergenic Regions</a><td>Takes a GFF file and exports regions not covered by any annotation, the intergenic regions.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ExportIntronicRegions">Export Intronic Regions</a><td>Takes a UCSC gene table and exports the most conservative estimate of intronic sequence.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#MergeUCSCGeneTable">Merge UCSC Gene Table</a><td>Merges transcript models that share the same gene name. Maximizes exons, minimizes introns.</tr>

<tr><td ID='cr' COLSPAN="2">&nbsp;</tr>
<tr><td ID='cr' COLSPAN="2">Misc Analysis Apps</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#Alleler">Alleler</a><td>Intersects a list of alleles (SNPs and INDELs) with gene models and returns their effects on coding sequences (synonymous/ non-synonymous/ frame shift) and splice-junctions.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#OligoTiler">Oligo Tiler</a><td>Tiles oligos across genomic regions returning their forward and reverse sequences. Good for creating microarray seq capture designs.</tr>	
<tr><td ID='rt'><a href="cmdLnMenus.html#Primer3Wrapper">Primer 3 Wrapper</a><td>Wrapper for the <a href="http://frodo.wi.mit.edu/primer3/">primer3 (http://frodo.wi.mit.edu/primer3/)</a> application. Extracts sequence, formats for primer3, executes, and parses the output to a spreadsheet. Useful for bulk qPCR primer picking. Yes, you do need to validate your results.</tr>	

<tr><td ID='cr' COLSPAN="2">&nbsp;</tr>
<tr><td ID='cr' COLSPAN="2">Converter Apps</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#MaqSnps2Bed">Maq Snps 2 Bed</a><td>Converts a Maq snp text file (1 based coordinates) into a bed file (interbase coordinates).</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#Text2USeq">Text 2 USeq</a><td>Converts genomic data text files (xxx.bed, xxx.gff, xxx.sgr, etc) to ultra compact, indexed, USeq binary files.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#Wig2USeq">Wig 2 USeq</a><td>Converts UCSC variable step, fixed step, and bedGraph xxx.wig/bedGraph4(.zip/.gz OK) files into stair step/ heat map useq archives.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#UCSCBig2USeq">UCSC Big 2 USeq</a><td>Converts UCSC bigWig (xxx.bw) or bigBed (xxx.bb) archives to USeq archives (xxx.useq).</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#USeq2UCSCBig">USeq 2 UCSC Big</a><td>Converts USeq archives to UCSC bigWig (xxx.bw) or bigBed (xxx.bb) archives based on the data type.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#USeq2Text">USeq 2 Text</a><td>Converts USeq binary files to native, bed, or wig formats.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#Bar2USeq">Bar 2 USeq</a><td>Converts directories of xxx.bar files into useq archives, recursive.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#Bar2Gr">Bar 2 Gr</a><td>Converts xxx.bar files to text xxx.gr files.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#Gr2Bar">Gr 2 Bar</a><td>Converts chromosome specific xxx.gr.zip files (position score) to binary chromosome specific xxx.bar files.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#Sgr2Bar">Sgr 2 Bar</a><td>Converts xxx.sgr.zip files (chr position score) to binary chromosome specific xxx.bar files. </tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#Bed2Bar">Bed 2 Bar</a><td>Converts xxx.bed files (chr start stop... name score strand) into binary chromosome specific xxx.bar file graphs. Can also generate a merged composite bed file thresholded at the sum of the overlapping bed scores.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#Wig2Bar">Wig 2 Bar</a><td>Converts variable step and fixed step xxx.wig(Var) files to chrom specific bar files.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#Graph2Bed">Graph 2 Bed</a><td>Converts USeq stair step and heat map graphs into region bed files using a threshold.</tr>

<tr><td ID='cr' COLSPAN="2">&nbsp;</tr>
<tr><td ID='cr' COLSPAN="2">Utility Apps</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#RandomizeTextFile">Randomize Text File</a><td>Randomizes the lines in a text file.  Good for randomly selecting reverse oligos to pad a microarray design.</tr>	
<tr><td ID='rt'><a href="cmdLnMenus.html#FileCrossFilter">File Cross Filter</a><td>FCF take a column in the matcher file and uses it to parse the rows from other files. Useful for pulling out and printing in order the rows that match the first file.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#FileMatchJoiner">File Match Joiner</a><td>FMJ loads a file and a particular column containing unique entries, a key, and then appends the key line to lines in the parsed file that match a particular column. Useful for appending chromosome coordinates to snp data based on a common ID, etc.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#FileJoiner">File Joiner</a><td>Joins many text files together, paying attention to avoid fusing the last and first lines from two files, a major headache with cat * >>	</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#FileSplitter">File Splitter</a><td>Splits a text file into many files containing a given number of lines.	</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#PrintSelectColumns">Print Select Columns</a><td>Spread sheet/ tab delimited file manipulations.</tr>

<tr><td ID='cr' COLSPAN="2">&nbsp;</tr>
<tr><td ID='cr' COLSPAN="2">Odd Alignment Parser Apps</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ElandParser">Eland Parser</a><td>Splits and converts stand alone or Solexa pipeline Eland Extended xxx_export.txt and xxx_sorted.txt alignment files into center position alignment scored binary PointData xxx.bar files.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ElandMultiParser">Eland Multi Parser</a><td>Parses an Eland xxx.eland_multi.txt file tabulating hits to each fasta entry.  Good for scoring hits to a transcriptome where every fasta entry represents a different gene.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#ElandSequencingParser">Eland Sequencing Parser</a><td>Generates sequencing summary tracks for each base and a called consensus from Eland export and sorted alignment files.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#NovoalignParser">Novoalign Parser</a><td>Parses Novoalign files into center position binary PointData xxx.bar files for USeq analysis and xxx.bed files.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#NovoalignPairedParser">Novoalign Paired Parser</a><td>Parses paired Novoalign files into 12 column xxx.bed files. Useful for visualizing sequenced fragments.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#NovoalignIndelParser">Novoalign Indel Parser</a><td>Parses Novoalign alignment files for consensus indels, something currently not supported by the maq apps. See Alleler app.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#SoapV1Parser">Soap V1 Parser</a><td>Parses a Soap version 1 alignment txt file into PointData, split by chromosome and strand.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#Tag2Point">Tag 2 Point</a><td>Splits and converts tab delimited text (chr start stop ... strand, e.g. xxx.bed) text files into center position binary xxx.bar PointData files used by ScanSeqs and IGB for analysis and visualization respectively.  Very fast loading.  Small size.</tr>
<tr><td ID='rt'><a href="cmdLnMenus.html#FilterDuplicateAlignments">Filter Duplicate Alignments</a><td>Filters identical reads from alignment files by random subsampling, finding the best, or limiting to a max number. Useful for minimizing PCR amplification artifacts in sequencing data.</tr>

</center>
</table>	
</body>
</html>