<html>
<head>
	<title>USeq Output File Type Descriptions</title>

<style type="text/css">
	#rt{text-align:right; color: #000000; font-weight: bold}
	#grBk {background-color: #CC9966;}
	TD {font-family: Verdana, Arial, Helvetica, sans-serif; font-size:12;}
	H1 {color: #996633; font:arial; font-size:16;}
	H2 {color: #996633; font:arial; font-size:12;}
	BODY {color:black; background-color:white; font-family: Verdana, Arial, Helvetica, sans-serif; font-size:12;}
	
	A:link    {text-decoration: none; color: #000000; font-weight: bold}  
	A:visited {text-decoration: none; color: #000000; font-weight: bold}   
	A:hover   {text-decoration: none; color: #FFCC66; font-weight: bold} 
	A:active  {text-decoration: none; color: #000000; font-weight: bold}   
</style>
</head>

<body>
<H1>USeq results typically include:</H1>
<ul> 
	<li> Note all pvals and fdrs are transformed (-10*log10(pval or fdr)) thus 13 = 5%, 20 = 1%, 30 = 0.1%.
	<p>
	<li> Three data folders containing:
	<p>
		<ul>
		<li><b>PointData</b> These folders contain mapped read data split by chromosome and strand.  For each read, its center position, and any associated score is saved.  These are used by various USeq applications to perform analyisis.  They can be viewed in <a href="http://bioserver.hci.utah.edu/BioInfo/index.php/Software:IGB">IGB</a> but be aware that PointData that share the same position are graphed on top of each other in <a href="http://bioserver.hci.utah.edu/BioInfo/index.php/Software:IGB">IGB</a>, not summed.
			<p>

		<li><b>Windows:</b> several different measurements related to the overlapping window scanning summary statistics created by <a href="cmdLnMenus.html#ScanSeqs">ScanSeqs</a>. These are best used for visualization in the <a href="http://bioserver.hci.utah.edu/BioInfo/index.php/Software:IGB">IGB</a>. For each summary score, two types of window representations maybe generated. 
							<b>HeatMap</b> is a window representation best viewed in <a href="http://bioserver.hci.utah.edu/BioInfo/index.php/Software:IGB">IGB</a> as a stair-step or heatmap. 
							<b>Point</b> window summaries are where the window score is assigned to the center position in the window. Both use the xxx.bar.zip binary format.
			<p>
			<ul>
			<li> <b>BinPVal</b>: -10Log10(p-values) - Bionomial p-values for each window calculated by comparing the number of treatment reads to the number of control reads.  Not multiple test corrected. Treat this as a variance corrected score for ranking purposes.
			<li> <b>QValFDR</b>: -10Log10(q-values) - Window level bionomial p-values converted into q-value FDRs using John Storey's R package. 
			<li> <b>EmpFDR</b>: -10Log10(FDR) - Empirical false discovery rates based on a control vs control null distribution.
			<li> <b>Sum, Sum+, Sum-</b>: A sum of the window point data, combine or for each strand, only provided when a treatment only analysis is run. 
			<li> <b>xxx.swi</b>: Serializes window object array generated by <a href="cmdLnMenus.html#ScanSeqs">ScanSeqs</a> and used by the <a href="cmdLnMenus.html#EnrichedRegionMaker">EnrichedRegionMaker</a> application.
			</ul>
			<p>
		<li><b>Enriched Regions:</b> Enriched Regions (or Reduced Regions) from the EnrichedRegionMaker are collapsed overlapping windows best used for downstream analysis. Several different sets of ERs can be created by specifying multiple thresholds or asking the <a href="cmdLnMenus.html#EnrichedRegionMaker">EnrichedRegionMaker</a> application to produce the top 100, 200, 400, etc. ERs. 
                			For each ER set, a folder (name_threshold_#ERs) is created containing the following.
			<p>
			<ul>
			<li> <b>xxx.xls</b>: a spreadsheet report, these are ordered by the the best window binomial p-value score, the most significant on top. Some or all of the following will be found:
				<ul>  
					<li> #Hyperlinks - if the Integrated Genome Browser is open, clicking these links will take you to that region in the genome browser.
					<li> Chr  - chromosome 
					<li> Start  - start bp coordinate (0 based) for the entire enriched (or reduced) region 
					<li> Stop - stop bp coordinate (end excluded) for the entire enriched (or reduced) region 
					<li> #Windows  - number of windows that were merged into the enriched region
					<li> #T - if PointData were provided, the number of treatment reads in the enriched region
					<li> #Unique_T - if PointData were provided, the number of unique treatment reads in the enriched region
					<li> #C - if PointData were provided, the number of control reads in the enriched region
					<li> #Unique_C - if PointData were provided, the number of unique control reads in the enriched region
					<li> ER_BinPVal - if PointData were provided, the binomial p-value for the entire enriched region. These are not multiple test corrected.
					<li> ER_Log2((#T+1)/(#C+1)) - if PointData were provided, the normalized log2 ratio for the enriched region.
					<li> BSW_Start - if PointData were provided, start bp coordinate of the best sub window
					<li> BSW_Stop - if PointData were provided, end bp coordinate of the best sub window
					<li> BSW_BinPVal - if PointData were provided, the binomial p-value for the best sub window. These are not multiple test corrected.
					<li> BW_Start - start bp coordinate for the best window from all the windows merged
					<li> BW_Stop - end bp coordinate for the best window from all the windows merged
					<li> BW_BinPVal  - binomal pvalue for the best window
					<li> BW_QValueFDR  - FDR estimation derived from the binomial pvalue using Storey's q-value method
					<li> BW_EmpFDR  - FDR estimation based on generating a null distribution with your input data,  only used with a static chIP-seq, skipped with a dynamic analysis
					<li> BW_SkewPVal  - Bonferroni corrected binomial p-value looking at whether there is a skew in the distribution of + vs - stranded reads, some folks insist this can be used to filter out false positives, in our hands it hurts more than helps and should be ignored
					<li> BW_Log2((sumT+1)/(sumC+1))  - normalized log2 ratio, a measure of enrichment
					<li> BW_SumT+   - number of reads in the best window for the treatment + strand 
					<li> BW_SumT-   - number of reads in the best window for the treatment - strand 
					<li> BW_SumC+   - number of reads in the best window for the control + strand   
					<li> BW_SumC-   - number of reads in the best window for the control - strand   
					<li> Info for entire run - genome version and total read counts
				</ul>
			
			<li> <b>xxx.egr</b>: a multiple score EGR format file for viewing in IGB.
			<li> <b>xxx.gff</b>: a GFF file 
			<li> <b>XXXbpSubWinData</b>: If treatment and control data were provided to the EnrichedRegionMaker, each ER is rescanned using a small window to identify the best peaks within the ER. These are included in the spreadsheet report as well as outputed as graph bar files, xxx.egr, and xxx.gff files.

			</ul>
			<p>
		<li><b>Defined Region Scan Seqs output:</b>: a spreadsheet containing the following:
			<ul>
					<li> #Name - if the Integrated Genome Browser is open, clicking these links will take you to that defined region in the genome browser.
					<li> Chr  - chromosome 
					<li> Strand - strand of the defined region
					<li> Start  - start bp coordinate (0 based) for the defined region
					<li> Stop - stop bp coordinate (end excluded) for the defined region
					<li> pVal - uncorrected binomial p-value
					<li> qValFDR - FDR estimation derived from the binomial pvalue using Storey's q-value method
					<li> eFDR - Empirical FDR estimation based on generating a null distribution with your control/input data,  only used with a static analysis, skipped with a dynamic analysis
					<li> pValSkew - Bonferroni corrected binomial p-value looking at whether there is a skew in the distribution of + vs - stranded reads, some folks insist this can be used to filter out false positives, in our hands it hurts more than helps and should be ignored
					<li> pValDiffDist - Bonferroni corrected chi-square test of independence p-value looking for differences in the distribution of reads between the exons for a particular define region, good for flagging alternative splicing 
					<li> TotalRegionBPs - total bps in defined region
					<li> Log2((sumT+1)/(sumC+1))  - normalized log2 ratio
					<li> tSumPlus   - number of reads from the treatment + strand 
					<li> tSumMinus   - number of reads from the treatment - strand
					<li> tRPKM - treatment reads per kb of interrogated region per total million mapped reads
					<li> cSumPlus   - number of reads from the control + strand   
					<li> cSumMinus   - number of reads from the control - strand   
					<li> cRPKM - control reads per kb of interrogated region per total million mapped reads
					<li> SpliceJunctions - numerous columns looking at the difference in splice junction reads that map to a particular defined region.  Only calculated when novoalign bed format alignments were provided to the DRSS.
					<li> Info for entire run - genome version and total read counts
					
			</ul>	

		
		
					<p>
		<li><b>Defined Region Differential Seq output:</b>: a spreadsheet containing the following:
			<ul>
			<li> #DisplayName - if the Integrated Genome Browser is open, clicking these hyper links will take you to that defined region/ gene in the genome browser.
		<li> Name - present if found in the gene table.
		<li> Chr  - chromosome 
		<li> Strand - strand of the defined region
		<li> Start  - start bp coordinate (0 based) for the defined region
		<li> Stop - stop bp coordinate (end excluded) for the defined region
		<li> TotalBPs - total base pairs of the defined region/ gene's exons. 
		<li> PVal_ConditionA_ConditionB - uncorrected negative binomial -10*log10(pval) from DESeq for the paired ConditionA vs ConditionB comparison
		<li> FDR_ConditionA_ConditionB - Benjamini-Hochberg -10*log10(FDR) estimation derived from the DESeq p-values
		<li> VarCorLg2Rto_ConditionA_ConditionB - Log2 ratio estimation (pseudo median of ConditionA varCorr values - pseudo median of ConditionB varCor values) for the degree of differential expression based on DESeq's variance corrected counts for each sample. 
		<li> SpliceChiPVal_ConditionA_ConditionB - Bonferroni corrected chi-square test of independence -10*log10(pval) looking for differences in the distribution of reads between the exons for a particular gene.  Replica counts are merged for this analysis.
		<li> SpliceMaxLg2Rto_ConditionA_ConditionB - The maximum observed log2Ratio for differential exonic splicing normalized for total gene counts.
		<li> SpliceMaxExon_Treatment_Control - The zero based index for which exon displayed the maximum log2Ratio splicing difference.
		<li> Counts_ConditionReplica - number of fragments from each replica that aligned to the defined region or gene exons.
		<li> VarCorCounts_ConditionReplica   - DESeq's variance corrected fragment counts for each replica that aligned to the defined region or gene exons.
		<li> FPKM_ConditionReplica - FPKM corrected fragment counts for each replica that aligned to the defined region or gene exons. FPKM = #fragments/TotalBPs/1000/#fragmentsInDataSet/1M
			
			</ul>	
		</ul>
		
</ul>


</body>

</html>
